pkg/local_workflows/output_workflow/writer.go

```diff
@@ -0,0 +1,187 @@
+	if config.GetBool(OUTPUT_CONFIG_KEY_NO_DEFAULT_WRITER) {
+		return nil
+	}
+
+	writer := &WriterEntry{
+		writer: &newLineCloser{
+			writer: outputDestination.GetWriter(),
+		},
+		mimeType:        DEFAULT_MIME_TYPE,
+		renderEmptyData: true,
+		name:            DEFAULT_WRITER,
+	}
+
+	if config.GetBool(OUTPUT_CONFIG_KEY_SARIF) {
+		writer.mimeType = SARIF_MIME_TYPE
+	}
+
+	if config.GetBool(OUTPUT_CONFIG_KEY_JSON) {
+		writer.mimeType = JSON_MIME_TYPE
+	}
+
+	if config.IsSet(OUTPUT_CONFIG_TEMPLATE_FILE) {
+		writer.templates = []string{config.GetString(OUTPUT_CONFIG_TEMPLATE_FILE)}
+	}
+
+	// if existing, use a mimetype lookup table to translate the mimetype of the default writer
+	if config.IsSet(OUTPUT_CONFIG_KEY_DEFAULT_WRITER_LUT) {
+		if lut, ok := config.Get(OUTPUT_CONFIG_KEY_DEFAULT_WRITER_LUT).(map[string]string); ok {
+			if translatedMimetype, ok := lut[writer.mimeType]; ok {
+				writer.mimeType = translatedMimetype
+			}
+		}
+	}
+
+	return writer
+}
+
+func GetWritersFromConfiguration(config configuration.Configuration, outputDestination iUtils.OutputDestination) WriterMap {
+	// resulting map of writers and their templates
+	writerMap := &writerMapImpl{
+		writers: map[string][]*WriterEntry{},
+	}
+
+	// currently the only used default writer is sarif
+	if tmp := getDefaultWriter(config, outputDestination); tmp != nil {
+		writerMap.writers[tmp.mimeType] = []*WriterEntry{tmp}
+	}
+
+	// default file writers
+	fileWriters := []FileWriter{
+		{
+			OUTPUT_CONFIG_KEY_SARIF_FILE,
+			SARIF_MIME_TYPE,
+			[]string{},
+			true,
+		},
+		{
+			OUTPUT_CONFIG_KEY_JSON_FILE,
+			JSON_MIME_TYPE,
+			[]string{},
+			true,
+		},
+	}
+
+	// use configured file writers if available
+	if tmp, ok := config.Get(OUTPUT_CONFIG_KEY_FILE_WRITERS).([]FileWriter); ok {
+		fileWriters = tmp
+	}
+
+	for _, fileWriter := range fileWriters {
+		if config.IsSet(fileWriter.NameConfigKey) {
+			mapEntry, ok := writerMap.writers[fileWriter.MimeType]
+			if !ok {
+				mapEntry = []*WriterEntry{}
+			}
+
+			mapEntry = append(mapEntry, &WriterEntry{
+				writer:          &delayedFileOpenWriteCloser{Filename: config.GetString(fileWriter.NameConfigKey)},
+				mimeType:        fileWriter.MimeType,
+				templates:       fileWriter.TemplateFiles,
+				renderEmptyData: fileWriter.WriteEmptyContent,
+				name:            fileWriter.NameConfigKey,
+			})
+
+			writerMap.writers[fileWriter.MimeType] = mapEntry
+		}
+	}
+
+	return writerMap
+}
+
+func applyTemplatesToWriters(supportedMimeTypes []MimeType2Template, writers WriterMap) map[string]*WriterEntry {
+	writerMap := map[string]*WriterEntry{}
+	for _, supported := range supportedMimeTypes {
+		mimetypeWriter := writers.PopWritersByMimetype(supported.mimetype)
+		if len(mimetypeWriter) == 0 {
+			continue
+		}
+
+		// add template to writer
+		for _, writer := range mimetypeWriter {
+			writer.templates = supported.templates
```

Comment:
Suggestion: This will override externally configured values. Let's add a check that there is no template assigned yet.